### ISSUE TITLE
Issue with global install

### ISSUE THREAD

**Issue by hameerabbasi** [node-lifx/lifx-lan-client#10]
When trying to install on `raspbian` as well as `openhabian`.

```
$ sudo npm install -g lifx-lan-client
npm WARN deprecated postinstall-build@5.0.3: postinstall-build's behavior is now built into npm! You should migrate off of postinstall-build and use the new `prepare` lifecycle script with npm 5.0.0 or greater.

> lifx-lan-client@1.0.1 postinstall /usr/lib/node_modules/lifx-lan-client
> postinstall-build lib

postinstall-build:
  This version of npm (6.9.0) may not be compatible with postinstall-build! There
  are outstanding bugs in certain versions of npm that prevent it from working with
  postinstall-build. See https://github.com/exogen/postinstall-build#bugs-in-npm
  for more information.

npm ERR! path /root/.npm/_logs
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall scandir
npm ERR! Error: EACCES: permission denied, scandir '/root/.npm/_logs'
npm ERR!  [Error: EACCES: permission denied, scandir '/root/.npm/_logs'] {
npm ERR!   stack: "Error: EACCES: permission denied, scandir '/root/.npm/_logs'",
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'scandir',
npm ERR!   path: '/root/.npm/_logs'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).
glob error [Error: EACCES: permission denied, scandir '/root/.npm/_logs'] {
  stack: "Error: EACCES: permission denied, scandir '/root/.npm/_logs'",
  errno: -13,
  code: 'EACCES',
  syscall: 'scandir',
  path: '/root/.npm/_logs'
}
npm ERR! path /root/.npm/_locks
npm ERR! code EACCES
npm ERR! errno -13
npm ERR! syscall mkdir
npm ERR! Error: EACCES: permission denied, mkdir '/root/.npm/_locks'
npm ERR!  [Error: EACCES: permission denied, mkdir '/root/.npm/_locks'] {
npm ERR!   stack: "Error: EACCES: permission denied, mkdir '/root/.npm/_locks'",
npm ERR!   errno: -13,
npm ERR!   code: 'EACCES',
npm ERR!   syscall: 'mkdir',
npm ERR!   path: '/root/.npm/_locks'
npm ERR! }
npm ERR!
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm ERR!
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
npm ERR! the command again as root/Administrator (though this is not recommended).
postinstall-build:
  Error: Command failed: sh -c "/usr/bin/node" "/usr/lib/node_modules/npm/bin/npm-cli.js" install --only=dev


npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! lifx-lan-client@1.0.1 postinstall: `postinstall-build lib`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the lifx-lan-client@1.0.1 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2019-07-13T10_01_50_796Z-debug.log
```

**Comment by hameerabbasi** [comment]
I have tried with just `-g`, with/without `sudo`, with/without `sudo -s` and with/without `sudo -i`

**Comment by ristomatti** [comment]
Ah crap! I've ran across this issue myself. It's related to node version you're using. When I created the ES2015 refactoring of node-lifx codebase, it was not yet designed for npm publishing in mind. As a workaround this should work at least on node 8.x if its an option for you to upgrade. 

We need to make a npm prepublish build step instead of a postinstall one to make this truly compatible. Or make the postinstall call compatible with the listed node versions. Yet another option would be to drop support for older node versions and use the source as is. But I'd rather keep the compatibility as people still seem to run very old versions of node for various reasons. 

@sawtatoes is this perhaps something you could easily do? I don't have time to look into this possibly until next week. Also last time I've published something to npm was years ago. Besides I don't have the npm publish token anyway...

**Comment by ristomatti** [comment]
The core issue is this:

postinstall-build: This version of npm (6.9.0) may not be compatible with postinstall-build! There are outstanding bugs in certain versions of npm that prevent it from working with postinstall-build. See https://github.com/exogen/postinstall-build#bugs-in-npm for more information.

**Comment by ristomatti** [comment]
It's possible to run node 8.x also on older Raspberry Pi's and Pi Zero but if I remember correctly there's no prebuilt package for it. You can follow the instructions here to install a version you want. https://dev.to/vorillaz/install-nodejs-on-a-raspberry-pi-4hd5

**Comment by ristomatti** [comment]
And for Model 3/3+ you will want to use nodesource packages so you can just use `apt` for the installation. Here's how to do that https://www.battlehillmedia.com/installing-node-js-and-npm-on-raspberry-pi-3/

**Comment by hameerabbasi** [comment]
It seems those steps install version 10.15.2 even on a freshly-flashed pi.

**Comment by hameerabbasi** [comment]
I think the Rasbpian repos themselves provide a newer version of node.

**Comment by hameerabbasi** [comment]
```
$ node -v
v12.6.0
```

It's a recent `node` version. 😕

**Comment by hameerabbasi** [comment]
`yarn` worked fine.

**Comment by ristomatti** [comment]
Ah I mixed up npm version with node version. Possibly upgrading npm could have fixed it. Or maybe it was just an misleading error message...

Good to hear you got it working. This is a bug to be fixed regardless. Thanks for reporting!

**Comment by hameerabbasi** [comment]
Upgrading `npm` and/or downgrading to Node 10 doesn't fix it. The yarn global install isn't really global. 🙁

**Comment by hameerabbasi** [comment]
It seems older `npm`/`node` versions are supported but later ones aren't.

**Comment by ristomatti** [comment]
Thanks for the additional info. Now when I think about it, I believe it was when I installed node 10 when I encountered this. As it was just for my own usage with Node-RED, I think I resolved it by just manually running `npm run build` inside the folder.

All in all, the compilation step adds unnecessary bulk on the package and makes the installation slower as it could be already transpiled in the npm package.

**Comment by ristomatti** [comment]
@hameerabbasi I had a spare moment to look into this. #11 hopefully fixes this. @Sawtaytoes needs to review/approve and publish to npm first though. Meanwhile you can try if it using `github:ristomatti/lifx-lan-client#improve-install` as the `package.json` version for `lifx-lan-client` fixes it for you.

**Comment by hameerabbasi** [comment]
@Sawtaytoes Is it possible to release, given a lot of people will experience this? I plan to release a HomeBridge plugin depending on this.

**Comment by Sawtaytoes** [comment]
Just published v1.0.2. It has changes from #11.
